### PR TITLE
Add support for code coverage

### PR DESCRIPTION
**Pull request by junghans**
See:
<https://codecov.io/github/junghans/espressopp?branch=coverage>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -23,19 +23,26 @@ before_install:
...
+    - compiler: clang
+      env: EXTERNAL=ON COVERAGE=ON #clang has no coverage support 
+
 script:
   - mkdir build && cd build && 
-    PATH="$HOME/bin:/usr/lib/ccache:$PATH" cmake -DEXTERNAL_BOOST=$EXTERNAL -DEXTERNAL_MPI4PY=$EXTERNAL .. && 
+    PATH="$HOME/bin:/usr/lib/ccache:$PATH" cmake -DEXTERNAL_BOOST=$EXTERNAL -DEXTERNAL_MPI4PY=$EXTERNAL ${COVERAGE:+-DUSE_GCOV=ON} .. && 
     make -j4 all ${BUILD_UG:+ug} &&
-    make test ARGS="-V" &&
+    make test ARGS="-V ${COVERAGE:+ -E pi_water}" &&
```
Comment (MrTheodor) [inline .travis.yml]: why?

**Comment by MrTheodor** [comment]
I was thinking about this tool, Espresso has it for a while. But is it really deal correctly with the templates, which are the core of ESPP ?

File: .travis.yml (hunk 1 above)
Comment (junghans) [inline .travis.yml]: The `pi_water` test runs too long with `-O0`, I will fix that in another pull request.

**Comment by junghans** [comment]
Sure codecov isn't perfect, but better than nothing and in addition we can upload the same data to other services like coverall.io as well, all of them use `gcov` as a starting point.

File: .travis.yml (hunk 1 above)
Comment (junghans) [inline .travis.yml]: See: https://travis-ci.org/junghans/espressopp/jobs/107892462

Comment (junghans) [inline .travis.yml]: See #54

**Pull request merged**